### ISSUE TITLE
Compatibility with Apple M3 pro Chips

### ISSUE THREAD

**Issue by hesamossanloo** [weecology/DeepForest#656]
**Please describe A feature you would like to be added.**
Support and compatibility of the PyTorch with Apple M3 pro Chips.

**Describe the solution you'd like**
Please update the training logic so that it is compatible when it is running on MPS devices (a.k.a Apple M chips). Atm I get the following error while training:
Code: 
m.trainer.fit(m)

Error:
"Cannot convert a MPS Tensor to float64 dtype as the MPS framework doesn't support float64. Please use float32 instead."

**Describe alternatives you've considered**
I couldn't find any solutions, so I had to train the model on Google Colab, which it worked.

```
**Additional context**
Traceback log:
TypeError                                 Traceback (most recent call last)
Cell In[252], [line 2](vscode-notebook-cell:?execution_count=252&line=2)
      [1](vscode-notebook-cell:?execution_count=252&line=1) start_time = time.time()
----> [2](vscode-notebook-cell:?execution_count=252&line=2) m.trainer.fit(m)
      [3](vscode-notebook-cell:?execution_count=252&line=3) print(f"--- Training on GPU: {(time.time() - start_time):.2f} seconds ---")

File [~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/trainer.py:543](https://file+.vscode-resource.vscode-cdn.net/Users/hesam.ossanloo/Projects/Startup/SkogApp/DeepForest/~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/trainer.py:543), in Trainer.fit(self, model, train_dataloaders, val_dataloaders, datamodule, ckpt_path)
    [541](https://file+.vscode-resource.vscode-cdn.net/Users/hesam.ossanloo/Projects/Startup/SkogApp/DeepForest/~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/trainer.py:541) self.state.status = TrainerStatus.RUNNING
    [542](https://file+.vscode-resource.vscode-cdn.net/Users/hesam.ossanloo/Projects/Startup/SkogApp/DeepForest/~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/trainer.py:542) self.training = True
--> [543](https://file+.vscode-resource.vscode-cdn.net/Users/hesam.ossanloo/Projects/Startup/SkogApp/DeepForest/~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/trainer.py:543) call._call_and_handle_interrupt(
    [544](https://file+.vscode-resource.vscode-cdn.net/Users/hesam.ossanloo/Projects/Startup/SkogApp/DeepForest/~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/trainer.py:544)     self, self._fit_impl, model, train_dataloaders, val_dataloaders, datamodule, ckpt_path
    [545](https://file+.vscode-resource.vscode-cdn.net/Users/hesam.ossanloo/Projects/Startup/SkogApp/DeepForest/~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/trainer.py:545) )

File [~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/call.py:44](https://file+.vscode-resource.vscode-cdn.net/Users/hesam.ossanloo/Projects/Startup/SkogApp/DeepForest/~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/call.py:44), in _call_and_handle_interrupt(trainer, trainer_fn, *args, **kwargs)
     [42](https://file+.vscode-resource.vscode-cdn.net/Users/hesam.ossanloo/Projects/Startup/SkogApp/DeepForest/~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/call.py:42)     if trainer.strategy.launcher is not None:
     [43](https://file+.vscode-resource.vscode-cdn.net/Users/hesam.ossanloo/Projects/Startup/SkogApp/DeepForest/~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/call.py:43)         return trainer.strategy.launcher.launch(trainer_fn, *args, trainer=trainer, **kwargs)
---> [44](https://file+.vscode-resource.vscode-cdn.net/Users/hesam.ossanloo/Projects/Startup/SkogApp/DeepForest/~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/call.py:44)     return trainer_fn(*args, **kwargs)
     [46](https://file+.vscode-resource.vscode-cdn.net/Users/hesam.ossanloo/Projects/Startup/SkogApp/DeepForest/~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/call.py:46) except _TunerExitException:
     [47](https://file+.vscode-resource.vscode-cdn.net/Users/hesam.ossanloo/Projects/Startup/SkogApp/DeepForest/~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/call.py:47)     _call_teardown_hook(trainer)

File [~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/trainer.py:579](https://file+.vscode-resource.vscode-cdn.net/Users/hesam.ossanloo/Projects/Startup/SkogApp/DeepForest/~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/trainer.py:579), in Trainer._fit_impl(self, model, train_dataloaders, val_dataloaders, datamodule, ckpt_path)
    [572](https://file+.vscode-resource.vscode-cdn.net/Users/hesam.ossanloo/Projects/Startup/SkogApp/DeepForest/~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/trainer.py:572) assert self.state.fn is not None
    [573](https://file+.vscode-resource.vscode-cdn.net/Users/hesam.ossanloo/Projects/Startup/SkogApp/DeepForest/~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/trainer.py:573) ckpt_path = self._checkpoint_connector._select_ckpt_path(
    [574](https://file+.vscode-resource.vscode-cdn.net/Users/hesam.ossanloo/Projects/Startup/SkogApp/DeepForest/~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/trainer.py:574)     self.state.fn,
...
---> [96](https://file+.vscode-resource.vscode-cdn.net/Users/hesam.ossanloo/Projects/Startup/SkogApp/DeepForest/~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/lightning_fabric/utilities/apply_func.py:96) data_output = data.to(device, **kwargs)
     [97](https://file+.vscode-resource.vscode-cdn.net/Users/hesam.ossanloo/Projects/Startup/SkogApp/DeepForest/~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/lightning_fabric/utilities/apply_func.py:97) if data_output is not None:
     [98](https://file+.vscode-resource.vscode-cdn.net/Users/hesam.ossanloo/Projects/Startup/SkogApp/DeepForest/~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/lightning_fabric/utilities/apply_func.py:98)     return data_output

TypeError: Cannot convert a MPS Tensor to float64 dtype as the MPS framework doesn't support float64. Please use float32 instead.
```

**Comment by bw4sz** [comment]
Can I see the full code and let's try to make this reproducible. I'm not sure I can make the jump to a particular chip from this error. As an aside, DeepForest runs on pytorch, which is probably the dominant open-source package for machine learning, we do not have any insight into the more fundamental hardware choices they make, but I also extremely doubt (but could be surprised!) if a major type of hardware would not be supported. This doesn't feel likely.

**Comment by hesamossanloo** [comment]
Sure, after splitting the training and test annotations, following the DeepForest docs, here are the configs:
```
m_skogapp.config['gpus'] = '-1'  # move to GPU and use all the GPU resources
m_skogapp.config["train"]["csv_file"] = 'train_annotations.csv'
m_skogapp.config["train"]["root_dir"] = os.path.dirname(
    'train_annotations.csv')
# Since this is a demo example and we aren't training for long, only show the higher quality boxes
m_skogapp.config["score_thresh"] = 0.4
m_skogapp.config["train"]['epochs'] = 5
m_skogapp.config["validation"]["csv_file"] = 'test_annotations.csv'
m_skogapp.config["validation"]["root_dir"] = os.path.dirname(
    'test_annotations.csv')
# create a pytorch lighting trainer used to training
m_skogapp.create_trainer()
# load the lastest release model
m_skogapp.use_release()
```
Here is the cell output:
```
GPU available: True (mps), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
`Trainer(limit_val_batches=1.0)` was configured so 100% of the batches will be used..
Model from DeepForest release https://github.com/weecology/DeepForest/releases/tag/1.0.0 was already downloaded. Loading model from file.
Loading pre-built model: https://github.com/weecology/DeepForest/releases/tag/1.0.0
```
And hre is the training code:
```
start_time = time.time()
m_skogapp.trainer.fit(m_skogapp)
print(f"--- Training on GPU: {(time.time() - start_time):.2f} seconds ---")
```
And here is when I get this error:
```
{
	"name": "TypeError",
	"message": "Cannot convert a MPS Tensor to float64 dtype as the MPS framework doesn't support float64. Please use float32 instead.",
	"stack": "---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[252], line 2
      1 start_time = time.time()
----> 2 m.trainer.fit(m)
      3 print(f\"--- Training on GPU: {(time.time() - start_time):.2f} seconds ---\")

File ~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/trainer.py:543, in Trainer.fit(self, model, train_dataloaders, val_dataloaders, datamodule, ckpt_path)
    541 self.state.status = TrainerStatus.RUNNING
    542 self.training = True
--> 543 call._call_and_handle_interrupt(
    544     self, self._fit_impl, model, train_dataloaders, val_dataloaders, datamodule, ckpt_path
    545 )

File ~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/call.py:44, in _call_and_handle_interrupt(trainer, trainer_fn, *args, **kwargs)
     42     if trainer.strategy.launcher is not None:
     43         return trainer.strategy.launcher.launch(trainer_fn, *args, trainer=trainer, **kwargs)
---> 44     return trainer_fn(*args, **kwargs)
     46 except _TunerExitException:
     47     _call_teardown_hook(trainer)

File ~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/trainer.py:579, in Trainer._fit_impl(self, model, train_dataloaders, val_dataloaders, datamodule, ckpt_path)
    572 assert self.state.fn is not None
    573 ckpt_path = self._checkpoint_connector._select_ckpt_path(
    574     self.state.fn,
    575     ckpt_path,
    576     model_provided=True,
    577     model_connected=self.lightning_module is not None,
    578 )
--> 579 self._run(model, ckpt_path=ckpt_path)
    581 assert self.state.stopped
    582 self.training = False

File ~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/trainer.py:986, in Trainer._run(self, model, ckpt_path)
    981 self._signal_connector.register_signal_handlers()
    983 # ----------------------------
    984 # RUN THE TRAINER
    985 # ----------------------------
--> 986 results = self._run_stage()
    988 # ----------------------------
    989 # POST-Training CLEAN UP
    990 # ----------------------------
    991 log.debug(f\"{self.__class__.__name__}: trainer tearing down\")

File ~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/trainer.py:1030, in Trainer._run_stage(self)
   1028 if self.training:
   1029     with isolate_rng():
-> 1030         self._run_sanity_check()
   1031     with torch.autograd.set_detect_anomaly(self._detect_anomaly):
   1032         self.fit_loop.run()

File ~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/trainer.py:1059, in Trainer._run_sanity_check(self)
   1056 call._call_callback_hooks(self, \"on_sanity_check_start\")
   1058 # run eval step
-> 1059 val_loop.run()
   1061 call._call_callback_hooks(self, \"on_sanity_check_end\")
   1063 # reset logger connector

File ~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/loops/utilities.py:182, in _no_grad_context.<locals>._decorator(self, *args, **kwargs)
    180     context_manager = torch.no_grad
    181 with context_manager():
--> 182     return loop_run(self, *args, **kwargs)

File ~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/loops/evaluation_loop.py:135, in _EvaluationLoop.run(self)
    133     self.batch_progress.is_last_batch = data_fetcher.done
    134     # run step hooks
--> 135     self._evaluation_step(batch, batch_idx, dataloader_idx, dataloader_iter)
    136 except StopIteration:
    137     # this needs to wrap the `*_step` call too (not just `next`) for `dataloader_iter` support
    138     break

File ~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/loops/evaluation_loop.py:370, in _EvaluationLoop._evaluation_step(self, batch, batch_idx, dataloader_idx, dataloader_iter)
    368     batch = trainer.precision_plugin.convert_input(batch)
    369     batch = trainer.lightning_module._on_before_batch_transfer(batch, dataloader_idx=dataloader_idx)
--> 370     batch = call._call_strategy_hook(trainer, \"batch_to_device\", batch, dataloader_idx=dataloader_idx)
    372 # the `_step` methods don't take a batch_idx when `dataloader_iter` is used, but all other hooks still do,
    373 # so we need different kwargs
    374 hook_kwargs = self._build_kwargs(
    375     batch, batch_idx, dataloader_idx if self._is_sequential and self.num_dataloaders > 1 else None
    376 )

File ~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/call.py:309, in _call_strategy_hook(trainer, hook_name, *args, **kwargs)
    306     return None
    308 with trainer.profiler.profile(f\"[Strategy]{trainer.strategy.__class__.__name__}.{hook_name}\"):
--> 309     output = fn(*args, **kwargs)
    311 # restore current_fx when nested context
    312 pl_module._current_fx_name = prev_fx_name

File ~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/strategies/strategy.py:278, in Strategy.batch_to_device(self, batch, device, dataloader_idx)
    276 device = device or self.root_device
    277 if model is not None:
--> 278     return model._apply_batch_transfer_handler(batch, device=device, dataloader_idx=dataloader_idx)
    279 return move_data_to_device(batch, device)

File ~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/core/module.py:347, in LightningModule._apply_batch_transfer_handler(self, batch, device, dataloader_idx)
    343 def _apply_batch_transfer_handler(
    344     self, batch: Any, device: Optional[torch.device] = None, dataloader_idx: int = 0
    345 ) -> Any:
    346     device = device or self.device
--> 347     batch = self._call_batch_hook(\"transfer_batch_to_device\", batch, device, dataloader_idx)
    348     batch = self._call_batch_hook(\"on_after_batch_transfer\", batch, dataloader_idx)
    349     return batch

File ~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/core/module.py:336, in LightningModule._call_batch_hook(self, hook_name, *args)
    333     else:
    334         trainer_method = call._call_lightning_datamodule_hook
--> 336     return trainer_method(trainer, hook_name, *args)
    337 hook = getattr(self, hook_name)
    338 return hook(*args)

File ~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/trainer/call.py:157, in _call_lightning_module_hook(trainer, hook_name, pl_module, *args, **kwargs)
    154 pl_module._current_fx_name = hook_name
    156 with trainer.profiler.profile(f\"[LightningModule]{pl_module.__class__.__name__}.{hook_name}\"):
--> 157     output = fn(*args, **kwargs)
    159 # restore current_fx when nested context
    160 pl_module._current_fx_name = prev_fx_name

File ~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/pytorch_lightning/core/hooks.py:613, in DataHooks.transfer_batch_to_device(self, batch, device, dataloader_idx)
    562 def transfer_batch_to_device(self, batch: Any, device: torch.device, dataloader_idx: int) -> Any:
    563     \"\"\"Override this hook if your :class:`~torch.utils.data.DataLoader` returns tensors wrapped in a custom data
    564     structure.
    565 
   (...)
    611 
    612     \"\"\"
--> 613     return move_data_to_device(batch, device)

File ~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/lightning_fabric/utilities/apply_func.py:102, in move_data_to_device(batch, device)
     99     # user wrongly implemented the `_TransferableDataType` and forgot to return `self`.
    100     return data
--> 102 return apply_to_collection(batch, dtype=_TransferableDataType, function=batch_to)

File ~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/lightning_utilities/core/apply_func.py:72, in apply_to_collection(data, dtype, function, wrong_dtype, include_none, allow_frozen, *args, **kwargs)
     70     return {k: function(v, *args, **kwargs) for k, v in data.items()}
     71 # slow path for everything else
---> 72 return _apply_to_collection_slow(
     73     data,
     74     dtype,
     75     function,
     76     *args,
     77     wrong_dtype=wrong_dtype,
     78     include_none=include_none,
     79     allow_frozen=allow_frozen,
     80     **kwargs,
     81 )

File ~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/lightning_utilities/core/apply_func.py:125, in _apply_to_collection_slow(data, dtype, function, wrong_dtype, include_none, allow_frozen, *args, **kwargs)
    123 out = []
    124 for d in data:
--> 125     v = _apply_to_collection_slow(
    126         d,
    127         dtype,
    128         function,
    129         *args,
    130         wrong_dtype=wrong_dtype,
    131         include_none=include_none,
    132         allow_frozen=allow_frozen,
    133         **kwargs,
    134     )
    135     if include_none or v is not None:
    136         out.append(v)

File ~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/lightning_utilities/core/apply_func.py:125, in _apply_to_collection_slow(data, dtype, function, wrong_dtype, include_none, allow_frozen, *args, **kwargs)
    123 out = []
    124 for d in data:
--> 125     v = _apply_to_collection_slow(
    126         d,
    127         dtype,
    128         function,
    129         *args,
    130         wrong_dtype=wrong_dtype,
    131         include_none=include_none,
    132         allow_frozen=allow_frozen,
    133         **kwargs,
    134     )
    135     if include_none or v is not None:
    136         out.append(v)

File ~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/lightning_utilities/core/apply_func.py:104, in _apply_to_collection_slow(data, dtype, function, wrong_dtype, include_none, allow_frozen, *args, **kwargs)
    102 out = []
    103 for k, v in data.items():
--> 104     v = _apply_to_collection_slow(
    105         v,
    106         dtype,
    107         function,
    108         *args,
    109         wrong_dtype=wrong_dtype,
    110         include_none=include_none,
    111         allow_frozen=allow_frozen,
    112         **kwargs,
    113     )
    114     if include_none or v is not None:
    115         out.append((k, v))

File ~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/lightning_utilities/core/apply_func.py:96, in _apply_to_collection_slow(data, dtype, function, wrong_dtype, include_none, allow_frozen, *args, **kwargs)
     84 def _apply_to_collection_slow(
     85     data: Any,
     86     dtype: Union[type, Any, Tuple[Union[type, Any]]],
   (...)
     93 ) -> Any:
     94     # Breaking condition
     95     if isinstance(data, dtype) and (wrong_dtype is None or not isinstance(data, wrong_dtype)):
---> 96         return function(data, *args, **kwargs)
     98     elem_type = type(data)
    100     # Recursively apply to collection items

File ~/Projects/Startup/SkogApp/DeepForest/.venv/lib/python3.11/site-packages/lightning_fabric/utilities/apply_func.py:96, in move_data_to_device.<locals>.batch_to(data)
     94 if isinstance(data, Tensor) and isinstance(device, torch.device) and device.type not in _BLOCKING_DEVICE_TYPES:
     95     kwargs[\"non_blocking\"] = True
---> 96 data_output = data.to(device, **kwargs)
     97 if data_output is not None:
     98     return data_output

TypeError: Cannot convert a MPS Tensor to float64 dtype as the MPS framework doesn't support float64. Please use float32 instead."
}
```

I have tried both with CPU and MPS on my Mac and it didn't work. But when I run the same code on Google Colab GPU, without changing anything, then it works :)

I am sure I am making an embarrasing mistake somewhere.

**Comment by bw4sz** [comment]
Okay, let me summarize my understanding here. MPS is an alternative to CUDA GPU chips, which seems to be a newer/less supported area of the ecosystem. This is genuinely upstream of us, even major teams like Segment-Anything are seeing some action here:

https://github.com/facebookresearch/segment-anything/issues/94

However, I don't think its quite that it isn't 'supported', that feels a bit too far. Its that the tensor types that are permissible are different, and some downstream process was ignorant of that. Let's see if we can probe this and find a solution.

Using prebuilt data and a fully reproducible example

```
from deepforest import main
from deepforest import get_data
import os

m = main.deepforest()
m.use_release()

m.config["train"]["csv_file"] = get_data("example.csv") 
m.config["train"]["root_dir"] = os.path.dirname(get_data("example.csv"))
m.config["train"]["fast_dev_run"] = True    
m.trainer.fit(m)
```

You are welcome to close if you don't have time/interest to explore, but it'll require a bit of effort on your paper to determine the example place and offer a change I am guessing that it is here: https://github.com/weecology/DeepForest/blob/6b1c199ce65f95dc74c9e79d1e2c686e784910b2/deepforest/dataset.py#L130C43-L130C48. We will need to look at the dtypes if the above example fails. As an aside, i'm not sure the config anticipates this hardware difference.

```
m_skogapp.config['gpus'] = '-1'  # move to GPU and use all the GPU resources
```

worries me, but may not matter.

**Comment by hesamossanloo** [comment]
Thank you @bw4sz for such quick response. I also agree with you in terms of the potential root cause of the issue. I am always happy to help. I do have the time, but I am not sure if I am smart enough to resolve it. I will definitely give it a go and keep you updated. 

In the meantime, please share anything that you think could be helpful. 

Cheers

**Comment by hesamossanloo** [comment]
Updates:
I tried the following suggestions, but  so far I had no success:
1. I tried to override the trainer function provided by deepforest, as suggested in the FAQ of deepforest Docs:
`m.trainer = None
from pytorch_lightning import Trainer

    trainer = Trainer(
        accelerator="gpu",
        strategy="ddp",
        devices=model.config["devices"],
        enable_checkpointing=False,
        max_epochs=model.config["train"]["epochs"],
        logger=comet_logger
    )
trainer.fit(m)`

And it didn't work for me.

2. I tried to enforce pytorch to use float32 by running this at the beginning of my script:
`torch.set_default_dtype(torch.float32)`

And it didn't work for me.

3. But finally by enforcing the float32 while loading the data in the dataset.py. I managed to get it working. I will create a PR :)
